components-part1/src/pages/Home.test.js

```diff
@@ -0,0 +1,80 @@
+import * as ReactRouter from 'react-router';
+import { shallow } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import Home from './Home';
+import Content from '../partials/Home/Content';
+import Footer from '../partials/Home/Footer';
+
+//we need to overwrite the useLocation function to writable
+//so that we can return a mock location object
+Object.defineProperty(ReactRouter, 'useLocation', {
+  value: jest.fn(),
+  configurable: true,
+  writable: true,
+});
+
+const setup = (props = {}) => {
+  const component = shallow(<Home {...props} />);
+  return component;
+};
+
+describe('Home', () => {
+  it('renders Search correctly when url without movieId query', () => {
+    const mockLocation_withoutMovieId = {
+      pathname: '/',
+      state: {},
+      key: '',
+      search: '',
+      hash: '',
+    };
+    const useLocationSpy = jest
+      .spyOn(ReactRouter, 'useLocation')
+      .mockReturnValue(mockLocation_withoutMovieId);
+    const wrapper = setup();
+    expect(useLocationSpy).toHaveBeenCalled();
+    expect(toJson(wrapper)).toMatchSnapshot();
```

Comment:
should clear spy after the test finished, as the same spy used in multiple tests.

```
useLocationSpy.mockClear()
```